pkg/applier/applier_test.go

```diff
@@ -391,6 +404,198 @@ func TestApply(t *testing.T) {
 	}
 }
 
+func TestApplyMutationIgnoredObjects(t *testing.T) {
+	rootSyncName := "my-rs"
+	syncScope := declared.RootScope
+	applySetID := applyset.IDFromSync(rootSyncName, syncScope)
+	testGitCommit := "example-commit"
+	gitContextOutput := fmt.Sprintf(`{"repo":%q,"branch":%q,"rev":%q}`,
+		"example-repo", "example-branch", testGitCommit)
+	resourceManager := declared.ResourceManager(syncScope, rootSyncName)
+
+	testcases := []struct {
+		name                       string
+		serverObjs                 []client.Object
+		declaredObjs               []client.Object
+		cachedIgnoredObjs          []client.Object
+		expectedObjsToApply        object.UnstructuredSet
+		expectedItemsInIgnoreCache []client.Object
+	}{
+		{
+			name: "unmanaged object exists on the cluster",
+			serverObjs: []client.Object{
+				createServerObject(
```

Comment:
Why wrapping the object with `createServerObject`? I don't think that is needed.